src/components/DateTimePicker.js

```diff
@@ -0,0 +1,292 @@
+      datePlaceholder: 'Select a Date',
+      locale: 'en',
+      onDateSelect () {},
+      primaryColor: StyleConstants.Colors.PRIMARY,
+      showIcons: true,
+      timeFormat: 'LT',
+      timeIcon: 'clock',
+      timePlaceholder: 'Select a Time',
+      timezone: moment.tz.guess(),
+      timezoneNames: {
+        EST: 'Eastern Standard Time',
+        EDT: 'Eastern Daylight Time',
+        CST: 'Central Standard Time',
+        CDT: 'Central Daylight Time',
+        MST: 'Mountain Standard Time',
+        MDT: 'Mountain Daylight Time',
+        PST: 'Pacific Standard Time',
+        PDT: 'Pacific Daylight Time'
+      }
+    };
+  },
+
+  getInitialState () {
+    return {
+      showCalendar: false,
+      editTime: false
+    };
+  },
+
+  _handleDateBlur () {
+    this.setState({
+      showCalendar: false
+    });
+  },
+
+  _handleDateFocus () {
+    this.setState({
+      showCalendar: true
+    });
+  },
+
+  _handleDateSelect (date, e) {
+    e.stopPropagation();
+
+    if (this.props.closeOnDateSelect) {
+      this.refs.dateSelect.blur();
+
+      this.setState({
+        showCalendar: false
+      });
+    }
+
+    const hour = this.props.selectedDate ? moment.unix(this.props.selectedDate).hour() : 0;
+    const minutes = this.props.selectedDate ? moment.unix(this.props.selectedDate).minute() : 0;
+
+    this.props.onDateSelect(moment.unix(date).hour(hour).minute(minutes).seconds(0).unix());
+  },
+
+  _handleTimeBlur () {
+    this.setState({
+      editTime: false
+    });
+  },
+
+  _handleTimeFocus () {
+    this.setState({
+      editTime: true
+    });
+  },
+
+  _handleTimeSelect (e) {
+    const selectedDate = this.props.selectedDate ? moment.unix(this.props.selectedDate) : moment();
+    const value = e.target.value.toLowerCase().replace('.', '');
+
+    const time = value.split(':');
+    const pm = value.includes('pm');
+    const hour = pm ? Number(time[0]) + 12 : Number(time[0]);
+    const minute = Number(time[1].substring(0, 2));
+    const date = selectedDate.hour(hour).minute(minute).second(0).unix();
+
+    this.setState({
+      editTime: false
+    });
+
+    this.props.onDateSelect(date);
+  },
+
+  _getTimezone (date) {
+    const timezoneAbbr = date ? moment.unix(date).tz(this.props.timezone).format('z') : moment().tz(this.props.timezone).format('z');
+
+    if (this.props.timezoneFormat === 'name') {
+      return this.props.timezoneNames[timezoneAbbr] || timezoneAbbr;
+    } else if (this.props.timezoneFormat === 'abbr') {
+      return timezoneAbbr;
+    } else {
+      return null;
+    }
+  },
+
+  render () {
+    const styles = this.styles();
+
+    return (
+      <div style={styles.component}>
+        <div
+          onBlur={this._handleDateBlur}
+          onClick={this._handleDateClick}
+          onFocus={this._handleDateFocus}
+          ref='dateSelect'
+          style={Object.assign({}, styles.selectWrapper, this.state.showCalendar ? styles.activeSelectWrapper : null)}
```

Comment:
Since we're only changing the color can we check this in the styles?
```
this.state.showCalendar ? this.props.primaryColor : StyleConstants.Colors.FOG
```
Just my personal opinion but it feels cleaner than having an additional style.